include/extractor/guidance/classification_data.hpp

```diff
@@ -18,63 +15,129 @@ namespace extractor
-    LOW_PRIORITY_ROAD // a road simply included for connectivity. Should be avoided at all cost
-};
+    // a class that behaves like a motorway (separated directions)
+    bool motorway_class : 1;
+    // all types of link classes
+    bool link_class : 1;
+    // a low priority class is a pure connectivity way. It can be ignored in multiple decisions
+    // (e.g. fork on a primary vs service will not happen)
+    bool may_be_ignored : 1;
+    // the road priority is used as an indicator for forks. If the roads are of similar priority
+    // (difference <=1), we can see the road as a fork. Else one of the road classes is seen as
+    // obvious choice
+    unsigned priority : 5;
 
-FunctionalRoadClass functionalRoadClassFromTag(std::string const &tag);
+    // default construction
+    RoadClassification()
+        : motorway_class(false), link_class(false), may_be_ignored(false), priority(0)
+    {
+    }
 
-inline bool isRampClass(const FunctionalRoadClass road_class)
-{
-    // Primary Roads and down are usually too small to announce their links as ramps
-    return road_class == FunctionalRoadClass::MOTORWAY_LINK ||
-           road_class == FunctionalRoadClass::TRUNK_LINK;
-}
+    RoadClassification(bool motorway_class, bool link_class, bool may_be_ignored, unsigned priority)
+        : motorway_class(motorway_class), link_class(link_class), may_be_ignored(may_be_ignored),
+          priority(priority)
+    {
+    }
 
-// Links are usually smaller than ramps, but are sometimes tagged
-// as MOTORWAY_LINK if they exit/enter a motorway/trunk road.
-inline bool isLinkClass(const FunctionalRoadClass road_class)
-{
-    return road_class == FunctionalRoadClass::MOTORWAY_LINK ||
-           road_class == FunctionalRoadClass::TRUNK_LINK ||
-           road_class == FunctionalRoadClass::PRIMARY_LINK ||
-           road_class == FunctionalRoadClass::SECONDARY_LINK ||
-           road_class == FunctionalRoadClass::TERTIARY_LINK;
-}
+    inline bool isMotorwayClass() const { return motorway_class && !link_class; }
+
+    inline bool isRampClass() const { return motorway_class && link_class; }
+
+    inline bool isLinkClass() const { return link_class; }
+
+    inline bool isLowPriorityRoadClass() const { return may_be_ignored; }
+
+    inline bool operator==(const RoadClassification &other) const
+    {
+        return motorway_class == other.motorway_class && link_class == other.link_class &&
+               may_be_ignored == other.may_be_ignored && priority == other.priority;
+    }
+
+    inline std::string toString() const
+    {
+        return std::string() + (motorway_class ? "motorway" : "normal") +
+               (link_class ? "_link" : "") + (may_be_ignored ? " ignorable" : " important") +
+               std::to_string(priority);
+    }
+
+    inline void set(const osmium::Way &way)
+    {
+        const static auto class_hash = [] {
+            std::unordered_map<std::string, unsigned> hash;
+            hash["motorway"] = 0;
```

Comment:
The problem with the enums is that this is a intermediate step (to test if the main thing still works).
In the end, it should move to the profile where we need to handle the osm-strings and these respective flags should be set by the tags.